### PR TITLE
Add utoipa annotation to edition endpoint

### PR DESCRIPTION
**Pull request by leovalais**
Also moves `RailjsonObject` into `editoast_schemas` under `InfraObject`.

> [!TIP]
> Review each commit individually.

### CODE REVIEW

File: editoast/src/infra_cache/mod.rs [hunk 1]
```diff
@@ -93,22 +93,22 @@ pub enum ObjectCache {
     NeutralSection(NeutralSection),
 }
 
-impl From<RailjsonObject> for ObjectCache {
-    fn from(railjson: RailjsonObject) -> Self {
+impl From<InfraObject> for ObjectCache {
+    fn from(railjson: InfraObject) -> Self {
```
Comment (woshilapin) [inline editoast/src/infra_cache/mod.rs]:
```suggestion
    fn from(infra_object: InfraObject) -> Self {
```

File: editoast/src/infra_cache/operation/create.rs [hunk 2]
```diff
@@ -3,51 +3,20 @@ use diesel::sql_types::BigInt;
...
-    Detector { railjson: Detector },
-    BufferStop { railjson: BufferStop },
-    Route { railjson: Route },
-    OperationalPoint { railjson: OperationalPoint },
-    Electrification { railjson: Electrification },
-}
 
 pub async fn apply_create_operation<'r>(
-    railjson_object: &'r RailjsonObject,
+    railjson_object: &'r InfraObject,
```
Comment (woshilapin) [inline editoast/src/infra_cache/operation/create.rs]:
```suggestion
    infra_object: &'r InfraObject,
```

File: editoast/src/infra_cache/operation/mod.rs [hunk 3]
```diff
@@ -41,17 +112,14 @@ pub enum CacheOperation {
 }
 
 impl CacheOperation {
-    pub fn try_from_operation(
-        operation: &Operation,
-        railjson_object: RailjsonObject,
-    ) -> Result<Self> {
+    pub fn try_from_operation(operation: &Operation, railjson_object: InfraObject) -> Result<Self> {
```
Comment (woshilapin) [inline editoast/src/infra_cache/operation/mod.rs]:
```suggestion
    pub fn try_from_operation(operation: &Operation, infra_object: InfraObject) -> Result<Self> {
```

File: editoast/src/views/infra/auto_fixes/mod.rs [hunk 4]
```diff
@@ -233,9 +234,8 @@ fn reduce_operation(
             } = update;
             debug_assert_eq!(railjson_object.get_id(), &obj_id);
             debug_assert_eq!(railjson_object.get_type(), obj_type);
-            railjson_object
-                .patch(&railjson_patch)
-                .map(|railjson_object: RailjsonObject| Operation::Create(Box::new(railjson_object)))
+            patch_infra_object(&railjson_object, &railjson_patch)
+                .map(|railjson_object: InfraObject| Operation::Create(Box::new(railjson_object)))
```
Comment (woshilapin) [inline editoast/src/views/infra/auto_fixes/mod.rs]:
```suggestion
                .map(|infra_object: InfraObject| Operation::Create(Box::new(infra_object)))
```

File: editoast/src/views/infra/auto_fixes/mod.rs [hunk 5]
```diff
@@ -233,9 +234,8 @@ fn reduce_operation(
             } = update;
             debug_assert_eq!(railjson_object.get_id(), &obj_id);
             debug_assert_eq!(railjson_object.get_type(), obj_type);
-            railjson_object
-                .patch(&railjson_patch)
-                .map(|railjson_object: RailjsonObject| Operation::Create(Box::new(railjson_object)))
+            patch_infra_object(&railjson_object, &railjson_patch)
```
Comment (woshilapin) [inline editoast/src/views/infra/auto_fixes/mod.rs]: This one will need to modify the initialization in the destructuring above.

```suggestion
            patch_infra_object(&infra_object, &railjson_patch)
```

**Review by woshilapin — APPROVED** [review]
Mostly some renaming, nice job.

**Review by flomonster — APPROVED** [review]
LGTM (reviewed + tested)

**Comment by woshilapin** [comment]
Test comment

**Review by woshilapin — APPROVED** [review]
Test approve

**Review by Uriel-Sautron — APPROVED** [review]
LGTM (tested)

File: front/src/applications/editor/types.ts [hunk 6]
```diff
@@ -23,6 +24,12 @@ export type MapState = {
   viewport: ViewState;
 };
 
+// EDITOAST OPERATIONS
+
+export type CreateOperation = Operation & { operation_type: 'CREATE' };
```
Comment (SharglutDev) [inline front/src/applications/editor/types.ts]: When narrowing the type like that, if I hover on `CreateOperation`, I have that : 

Capture d’écran 2024-06-14 à 16 20 16

We can see that `operation_type: 'CREATE'` appears two times. It doesn't result to any error as we can't assign two times the same property in the object, but a moree elegant way would be to define it like that : 

```suggestion
export type CreateOperation = Extract<Operation, { operation_type: 'CREATE' }>;
```

File: front/src/applications/editor/types.ts [hunk 7]
```diff
@@ -23,6 +24,12 @@ export type MapState = {
...
+export type UpdateOperation = Operation & { operation_type: 'UPDATE' };
+export type DeleteOperation = Operation & { operation_type: 'DELETE' };
```
Comment (SharglutDev) [inline front/src/applications/editor/types.ts]: Same thing for these two

File: front/src/applications/editor/data/utils.ts [hunk 8]
```diff
@@ -149,15 +145,15 @@ export function nestEntity(entity: EditorEntity, type: EditoastType): EditorEnti
...
+export function entityToCreateOperation(entity: EditorEntity): CreateOperation {
   return {
     operation_type: 'CREATE',
     obj_type: entity.objType,
     railjson: {
       ...entity.properties,
       id: uuid(),
     },
-  };
+  } as unknown as CreateOperation;
```
Comment (SharglutDev) [inline front/src/applications/editor/data/utils.ts]: It seems to work without `as unknown `

```suggestion
  } as CreateOperation;
```

**Review by SharglutDev — CHANGES_REQUESTED** [review]
Thank you for the PR, tested it works great. Just left some comments.

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested

**Pull request merged**